### ISSUE TITLE
[Package Issue]: PushPlayLabs.Sidekick

### ISSUE THREAD

**Issue by jkruza** [microsoft/winget-pkgs#140742]
### Please confirm these before moving forward

- [X] I have searched for my issue and not found a work-in-progress/duplicate/resolved issue.
- [X] I have not been informed if the issue is resolved in a preview version of the winget client.

### Category of the issue

Other

### Brief description of your issue

Existing installation of another software with almost same name ("Sidekick" by Gigabyte) triggers attempt to "upgrade" it to  PushPlayLabs.Sidekick resulting in removing Gigabyte Sidekick and installing this package.

### Steps to reproduce

1. Install Gigabyte Sidekick (it is pre-installed on some Gigabyte Laptops or can be installed as user)
2. `winget upgrade`  - detects it as "new version". 
3. `winget upgrade --all` - removed Gigabyte Sidekick and installs PushPlayLabs.Sidekick

### Actual behavior

Winget replaces "Gigabyte Sidekick" with PushPlayLabs.Sidekick.

### Expected behavior

winget does understand this is different software and does not attempt to replace it.

### Environment

```raw
Windows Package Manager v1.6.3482
Copyright (c) Microsoft Corporation. All rights reserved.

Windows: Windows.Desktop v10.0.22631.3155
System Architecture: X64
Package: Microsoft.DesktopAppInstaller v1.21.3482.0

Winget Directories
-----------------------------------------------------------------------------------------------------------------------
Logs                               %LOCALAPPDATA%\Packages\Microsoft.DesktopAppInstaller_8wekyb3d8bbwe\LocalState\Diag…
User Settings                      %LOCALAPPDATA%\Packages\Microsoft.DesktopAppInstaller_8wekyb3d8bbwe\LocalState\sett…
Portable Links Directory (User)    %LOCALAPPDATA%\Microsoft\WinGet\Links
Portable Links Directory (Machine) C:\Program Files\WinGet\Links
Portable Package Root (User)       %LOCALAPPDATA%\Microsoft\WinGet\Packages
Portable Package Root              C:\Program Files\WinGet\Packages
Portable Package Root (x86)        C:\Program Files (x86)\WinGet\Packages
Installer Downloads                %USERPROFILE%\Downloads

Links
---------------------------------------------------------------------------
Privacy Statement   https://aka.ms/winget-privacy
License Agreement   https://aka.ms/winget-license
Third Party Notices https://aka.ms/winget-3rdPartyNotice
Homepage            https://aka.ms/winget
Windows Store Terms https://www.microsoft.com/en-us/storedocs/terms-of-sale

Admin Setting                             State
--------------------------------------------------
LocalManifestFiles                        Disabled
BypassCertificatePinningForMicrosoftStore Disabled
InstallerHashOverride                     Disabled
LocalArchiveMalwareScanOverride           Disabled
```

### Screenshots and Logs

Installed Gigabyte Sidekick:
![image](https://github.com/microsoft/winget-pkgs/assets/30356421/2b492d7f-946d-48b3-9b17-0175cf5fcd3f)

Running `winget --upgrade`

![image](https://github.com/microsoft/winget-pkgs/assets/30356421/95ec5394-851f-462c-b1b2-a127fa9c24a8)

**Comment by jkruza** [comment]
Additional info:
Gigabyte Sidekick is part of "GIGABYTE Control Center Full Installation Package" that can be found here:
https://www.gigabyte.com/Support/Utility?p=1

**Comment by amoisis** [comment]
I agree, this is effecting myself. seems to be an issue with how the winget upgrade command is detecting the installed applications

**Comment by abarela13** [comment]
I thought I was going insane looking for how this kept getting installed.

**Comment by julian-alarcon** [comment]
I still have this issue, can you fix this?

**Comment by Hanuwa** [comment]
This issue continues to be a significant frustration due to the lack of an option to exclude specific applications from upgrades. As a result, I am forced to install PushPlayLabs.Sidekick every time I want to update all of my applications at once

**Comment by SpecterShell** [comment]
> This issue continues to be a significant frustration due to the lack of an option to exclude specific applications from upgrades. As a result, I am forced to install PushPlayLabs.Sidekick every time I want to update all of my applications at once

You can try `winget pin add PushPlayLabs.Sidekick` to prevent it from being installed.

**Comment by arkimber** [comment]
Is the fix at https://github.com/microsoft/winget-pkgs/pull/140875 not available now then? It says closed but this bug is still open?

**Comment by SpecterShell** [comment]
> Is the fix at [#140875](https://github.com/microsoft/winget-pkgs/pull/140875) not available now then? It says closed but this bug is still open?

Nope. The PR was closed instead of being merged, which means the fix is not and will never be available.

**Comment by arkimber** [comment]
> > Is the fix at [#140875](https://github.com/microsoft/winget-pkgs/pull/140875) not available now then? It says closed but this bug is still open?
> 
> Nope. The PR was closed instead of being merged, which means the fix is not and will never be available.

Ah sorry to hear that, I saw that you did a lot of work on that. 🙁

Thanks also for the pin tip that removes it from the update all command. That looks like the only solution at this point.

**Comment by TheStig-1981** [comment]
> > This issue continues to be a significant frustration due to the lack of an option to exclude specific applications from upgrades. As a result, I am forced to install PushPlayLabs.Sidekick every time I want to update all of my applications at once
> 
> You can try `winget pin add PushPlayLabs.Sidekick` to prevent it from being installed.

Thank you. This works for me.
